js/errors.ts

```diff
@@ -1,17 +1,94 @@
+  const anonymousClass = class extends DenoError<T> {
+    constructor(msg: string) {
+      super(kind, msg);
+    }
+  };
+  Object.defineProperty(anonymousClass, "name", {
+    value: name,
+  });
+  errorClasses.set(kind, anonymousClass);
+  return anonymousClass;
+}
+
+// tslint:disable:variable-name
+
+// IO errors.
+export const ErrNotFound = ErrorFactory(fbs.ErrorKind.NotFound);
+export const ErrPermissionDenied = ErrorFactory(fbs.ErrorKind.PermissionDenied);
+export const ErrConnectionRefused = ErrorFactory(
+  fbs.ErrorKind.ConnectionRefused
+);
+export const ErrConnectionReset = ErrorFactory(fbs.ErrorKind.ConnectionReset);
+export const ErrConnectionAborted = ErrorFactory(
+  fbs.ErrorKind.ConnectionAborted
+);
+export const ErrNotConnected = ErrorFactory(fbs.ErrorKind.NotConnected);
+export const ErrAddrInUse = ErrorFactory(fbs.ErrorKind.AddrInUse);
+export const ErrAddrNotAvailable = ErrorFactory(fbs.ErrorKind.AddrNotAvailable);
+export const ErrBrokenPipe = ErrorFactory(fbs.ErrorKind.BrokenPipe);
+export const ErrAlreadyExists = ErrorFactory(fbs.ErrorKind.AlreadyExists);
+export const ErrWouldBlock = ErrorFactory(fbs.ErrorKind.WouldBlock);
+export const ErrInvalidInput = ErrorFactory(fbs.ErrorKind.InvalidInput);
+export const ErrInvalidData = ErrorFactory(fbs.ErrorKind.InvalidData);
+export const ErrTimedOut = ErrorFactory(fbs.ErrorKind.TimedOut);
+export const ErrInterrupted = ErrorFactory(fbs.ErrorKind.Interrupted);
+export const ErrWriteZero = ErrorFactory(fbs.ErrorKind.WriteZero);
+export const ErrOther = ErrorFactory(fbs.ErrorKind.Other);
+export const ErrUnexpectedEof = ErrorFactory(fbs.ErrorKind.UnexpectedEof);
+
+// URL errors.
+export const ErrEmptyHost = ErrorFactory(fbs.ErrorKind.EmptyHost);
+export const ErrIdnaError = ErrorFactory(fbs.ErrorKind.IdnaError);
+export const ErrInvalidPort = ErrorFactory(fbs.ErrorKind.InvalidPort);
+export const ErrInvalidIpv4Address = ErrorFactory(
+  fbs.ErrorKind.InvalidIpv4Address
+);
+export const ErrInvalidIpv6Address = ErrorFactory(
+  fbs.ErrorKind.InvalidIpv6Address
+);
+export const ErrInvalidDomainCharacter = ErrorFactory(
+  fbs.ErrorKind.InvalidDomainCharacter
+);
+export const ErrRelativeUrlWithoutBase = ErrorFactory(
+  fbs.ErrorKind.RelativeUrlWithoutBase
+);
+export const ErrRelativeUrlWithCannotBeABaseBase = ErrorFactory(
+  fbs.ErrorKind.RelativeUrlWithCannotBeABaseBase
+);
+export const ErrSetHostOnCannotBeABaseUrl = ErrorFactory(
+  fbs.ErrorKind.SetHostOnCannotBeABaseUrl
+);
+
+// Hyper errors.
+export const ErrOverflow = ErrorFactory(fbs.ErrorKind.Overflow);
+export const ErrHttpUser = ErrorFactory(fbs.ErrorKind.HttpUser);
+export const ErrHttpClosed = ErrorFactory(fbs.ErrorKind.HttpClosed);
+export const ErrHttpCanceled = ErrorFactory(fbs.ErrorKind.HttpCanceled);
+export const ErrHttpParse = ErrorFactory(fbs.ErrorKind.HttpParse);
+export const ErrHttpOther = ErrorFactory(fbs.ErrorKind.HttpOther);
+// tslint:enable:variable-name
```

Comment:
I'd very much like to avoid having multiple lists of these errors - but maybe having one is okay for now.

Could you add a test js/errors_test.ts which iterates over the values in fbs.ErrorKind and checks that there is a corresponding error class?